### PR TITLE
MM-22706 Don't set user online when responding to push notification

### PR DESCRIPTION
**Pull request by hmhealey**
The new set_online flag is passed from the server when a post is created with the `set_online` flag set to false, like when replying to a push notification from new versions of the mobile app.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-22706

### CODE REVIEW

**Comment by amyblais** [comment]
@hmhealey  Was this assigned to v5.24 by accident?

File: actions/websocket_actions.jsx [hunk 1]
```diff
@@ -539,7 +539,17 @@ export function handleNewPostEvent(msg) {
...
 
-        if (post.user_id !== getCurrentUserId(myGetState()) && !fromAutoResponder(post) && !getIsManualStatusForUserId(myGetState(), post.user_id)) {
+        // Since status updates aren't real time, assume another user is online if they have posted and:
+        // 1. The post isn't from the auto responder
+        // 2. The user hasn't set their status manually to something that isn't online
+        // 3. The server hasn't told the client not to set the user to online (because the other user
+        //    responded from a post notification)
+        if (
+            post.user_id !== getCurrentUserId(myGetState()) &&
+            !fromAutoResponder(post) &&
```
Comment (agnivade) [inline actions/websocket_actions.jsx:549]: I think now posts from autoResponder will have set_online to false. Is there still a need for this ?

**Comment by hmhealey** [comment]
@amyblais Whoops, misread the fix version in Jira. Thanks for fixing that.

File: actions/websocket_actions.jsx (hunk 1 above)
Comment (hmhealey) [reply actions/websocket_actions.jsx:549]: Right, I asked for that to be changed, didn't I? I'll remove the special handling for the auto responder then

**Comment by hmhealey** [comment]
@agnivade @catalintomai Can you guys take a look at this when you get a chance?

**Pull request merged**